packages/react/menu/src/Menu.tsx

```diff
@@ -428,44 +681,80 @@ const MenuItem = React.forwardRef((props, forwardedRef) => {
+
+const SUB_TRIGGER_NAME = 'MenuSubTrigger';
+const SUB_TRIGGER_DEFAULT_TAG = 'button';
+
+type MenuSubTriggerOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof MenuItemImpl>,
+  Polymorphic.OwnProps<typeof MenuAnchor>
+>;
+type MenuSubTriggerPrimitive = Polymorphic.ForwardRefComponent<
+  typeof SUB_TRIGGER_DEFAULT_TAG,
+  MenuSubTriggerOwnProps
+>;
+
+const MenuSubTrigger = React.forwardRef((props, forwardedRef) => {
+  const { disabled, as = SUB_TRIGGER_DEFAULT_TAG, ...triggerProps } = props;
+  const context = useMenuContext(SUB_TRIGGER_NAME);
+  const subMenuContext = useMenuSubContext(SUB_TRIGGER_NAME);
+  const [mouseInteracting, setMouseInteracting] = React.useState(false);
+
+  return (
+    <MenuItemImpl as={Slot} disabled={disabled}>
+      <MenuAnchor
+        type="button"
+        aria-haspopup="menu"
+        aria-expanded={context.open ? true : undefined}
+        aria-controls={context.open ? subMenuContext.contentId : undefined}
+        {...triggerProps}
+        as={as}
+        ref={composeRefs(forwardedRef, subMenuContext.onTriggerRef)}
+        data-state={getOpenState(context.open)}
+        onMouseMove={composeEventHandlers(triggerProps.onMouseMove, (event) => {
+          if (!disabled && !mouseInteracting && !context.open) {
+            setMouseInteracting(true);
+            event.currentTarget.focus();
           }
         })}
-        // we handle selection on `mouseUp` rather than `click` to match native menus implementation
-        onMouseUp={composeEventHandlers(props.onMouseUp, handleSelect)}
+        onMouseLeave={composeEventHandlers(triggerProps.onMouseLeave, (event) => {
+          setMouseInteracting(false);
+        })}
         /**
-         * We focus items on `mouseMove` to achieve the following:
-         *
-         * - Mouse over an item (it focuses)
-         * - Leave mouse where it is and use keyboard to focus a different item
-         * - Wiggle mouse without it leaving previously focused item
-         * - Previously focused item should re-focus
-         *
-         * If we used `mouseOver`/`mouseEnter` it would not re-focus when the mouse
-         * wiggles. This is to match native menu implementation.
+         * We hook into focus to control menu visibility rather than mouse events
+         * This solves a sticky open state problem when certain browser controls (e.g. devtools inspect overlay) are given focus
+         * This is because mouse events continue to fire while `onDismiss` in `DismissableLayer` won't get called due to it being a focus outside check
          */
-        onMouseMove={composeEventHandlers(props.onMouseMove, (event) => {
-          if (!disabled) {
-            const item = event.currentTarget;
-            item.focus();
-          } else {
-            contentContext.onItemLeave();
+        onFocus={composeEventHandlers(triggerProps.onFocus, (event) => {
+          if (mouseInteracting) {
+            subMenuContext.onMenuOpen(event);
```

Comment:
The `mouseInteracting` dance reads a little strange but the inline comment should clarify. I'm basically qualifying the interaction behind `focus` because of the way the layering is designed. Without doing this, when using inspect tools the menus will appear but not dismiss which is annoying AF.

Reply:
Would you mind showing me what the issue was to help me understand it a bit better?